plugin.image.daily_dilbert/addon.py

```diff
@@ -0,0 +1,321 @@
+# global variables including used Kodi arguments, which are not supposed to be changed or altered
+g_AddonHandle = int(sys.argv[1])
+g_AddonPath = xbmcaddon.Addon().getAddonInfo('path')
+g_AddonName = xbmcaddon.Addon().getAddonInfo('name')
+g_Args = parse_qs(sys.argv[2][1:])
+g_Args_Mode = g_Args.get('mode', None)
+g_Args_Year = g_Args['year'][0] if 'year' in g_Args else "";
+g_Args_Month= g_Args['month'][0] if 'month' in g_Args else "";
+g_Args_Day = g_Args['day'][0] if 'day' in g_Args else "";
+g_Args_Page = g_Args['page'][0] if 'page' in g_Args else "";
+g_Args_URL = sys.argv[0]
+g_CacheDir = g_AddonPath + "/cache/"									# Where to we store cached scraped Dilbert URLs
+g_Now = datetime.date.today()											# Today
+
+### Dilbert Settings ###
+g_UseCache = True														# Use the cache function
+g_PageItems = 7 														# the page_number items will be pre-fetched and cached, so the higher the number, the longer it takes.
+g_PageItemsRandom = 7													# items to be loaded/pre-scraped in random menu
+### Dilbert Scraper Configuration
+g_BaseUrl = "http://www.dilbert.com/strip/"								# Base URL, date will be added in form .../strip/yyy-mm-dd to get according Dilbert webpage
+g_Pattern = re.compile('"(http://assets.amuniversal.com/[a-z0-9]+)"')	# Pattern to search on Dilberts webpage
+g_FirstDilbert = datetime.date(1989,4,16) 								# 1. Dilbert online available on 16.4.1989, required for date checks
+g_cachemodefile='lastmode.cache'
+g_cacherandomdates='randomdates.cache'
+### Dilbert icons and fanart
+g_Icons={}
+g_Icons['today'] = g_AddonPath + "/resources/media/dil_today.png"
+g_Icons['recent'] = g_AddonPath + "/resources/media/dil_recent.png"
+g_Icons['random'] = g_AddonPath + "/resources/media/dil_random.png"
+g_Icons['date'] = g_AddonPath + "/resources/media/dil_bydate.png"
+g_Icons['browse'] = g_AddonPath + "/resources/media/dil_browse.png"
+g_Icons['next'] = g_AddonPath + "/resources/media/dil_next.png"
+g_Icons['click'] = g_AddonPath + "/resources/media/dil_click2show.png"	# For necessary click action to show strip as no preview is available
+g_FanartImage=["","","","","",""]
+g_FanartImage[0]= g_AddonPath + "/resources/media/fanart1.jpg"			# Unused for now
+g_FanartImage[1]= g_AddonPath + "/resources/media/fanart2.jpg"			# For "Today"
+g_FanartImage[2]= g_AddonPath + "/resources/media/fanart3.jpg"			# For "Random"
+g_FanartImage[3]= g_AddonPath + "/resources/media/fanart4.jpg"			# For "Recent"
+g_FanartImage[4]= g_AddonPath + "/resources/media/fanart5.jpg"			# For "Browse"
+g_FanartImage[5]= g_AddonPath + "/resources/media/fanart6.jpg"			# For "By Date"
+
+
+### SubRoutines ###
+def build_url(query):
+	return g_Args_URL + '?' + urllib.urlencode(query)
+
+def add_directory(mode=None,year='',month='',day='', page='',name='',icon='',fanart=''):
+	url = build_url({'mode': mode, 'foldername': name, 'page': page, 'year': year, 'month': month, 'day': day})
+	li = xbmcgui.ListItem(name, iconImage=icon)
```

Comment:
Thanks for finding this. I hope I fixed it correctly by using listitem.setIconImage(image) instead (at least it works)